src/main/java/org/tron/core/services/RpcApiService.java

```diff
@@ -294,6 +335,34 @@ public void getBlockByNum(NumberMessage request, StreamObserver<Block> responseO
       responseObserver.onCompleted();
     }
 
+    @Override
+    public void getBlockByNum2(NumberMessage request,
+        StreamObserver<BlockExtention> responseObserver) {
+      long num = request.getNum();
+      if (num >= 0) {
+        Block reply = wallet.getBlockByNum(num);
+        responseObserver.onNext(block2Extention(reply));
+      } else {
+        responseObserver.onNext(null);
+      }
+      responseObserver.onCompleted();
+    }
+
+    @Override
+    public void getTransactionCountByBlockNum(NumberMessage request,
+        StreamObserver<NumberMessage> responseObserver) {
+      Block block = null;
+      try {
+        block = dbManager.getBlockByNum(request.getNum()).getInstance();
+      } catch (StoreException e) {
+        logger.error(e.getMessage());
+      }
+      NumberMessage.Builder builder = NumberMessage.newBuilder();
+      builder.setNum(block.getTransactionsCount());
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "block" is nullable here. ([squid:S2259](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2259))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1180&issues=AWTWJrjdVtLgEGHvwB6I&open=AWTWJrjdVtLgEGHvwB6I)

[ISSUE_KEY]: AWTWJrjdVtLgEGHvwB6I